modules/cnv-automating-virtual-machine-creation-with-ansible.adoc

```diff
@@ -0,0 +1,98 @@
+// Module included in the following assemblies:
+//
+// * cnv/cnv_users_guide/cnv-automating-management-tasks.adoc
+
+[id="cnv-automating-virtual-machine-creation-with-ansible_{context}"]
+= Automating virtual machine creation with Red Hat Ansible Automation
+
+You can use the `kubevirt_vm` Ansible Playbook to create virtual machines in
+your {product-title} cluster.
+
+.Prerequisites
+
+* link:https://access.redhat.com/products/red-hat-ansible-engine[Red Hat Ansible Engine]
+version 2.8 or newer
+
+.Procedure
+
+. Edit an Ansible Playbook YAML file so that it includes the `kubevirt_vm` task:
++
+[source,yaml]
+----
+  kubevirt_vm:
+    namespace:
+    name:
+    cpu_cores:
+    memory:
+    disks:
+      - name:
+        volume:
+          containerDisk:
+            image:
+        disk:
+          bus:
+----
++
+[NOTE]
+====
+This snippet only includes the `kubevirt_vm` portion of the playbook.
+====
+
+. Edit the values to reflect the virtual machine you want to create, including the
+`namespace`, the number of `cpu_cores`, the `memory`, and the `disks`. For
+example:
++
+[source,yaml]
+----
+  kubevirt_vm:
+    namespace: default
+    name: vm1
+    cpu_cores: 1
+    memory: 64Mi
+    disks:
+      - name: containerdisk
+        volume:
+          containerDisk:
+            image: kubevirt/cirros-container-disk-demo:latest
+        disk:
+          bus: virtio
+----
+
+. If you want the virtual machine to boot immediately after creation, add
+`state: running` to the YAML file. For example:
++
+[source,yaml]
+----
+  kubevirt_vm:
+    namespace: default
+    name: vm1
+    state: running <1>
+    cpu_cores: 1
+----
+<1> Changing this value to `state: absent` deletes the virtual machine, if it
+already exists.
+
+.  Run the `ansible-playbook` command, using your playbook's file name as the
+only argument:
++
+----
+$ ansible-playbook create-vm.yaml
+----
+
+. Review the output to determine if the play was successful:
++
+----
+(...)
+TASK [Create my first VM] ************************************************************************
+changed: [localhost]
+
+PLAY RECAP ********************************************************************************************************
+localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
+----
+
+. If you did not include `state: running` in your playbook file, you can boot
+the virtual machine by running the playbook again:
```

Comment:
I think it could be made a little more explicit, that booting the virtual machine by running the playbook again is possible only after adding that `state: running` line.

Reply:
Hey - the source material says "Changing this behavior requires specifying state: running as one of the module’s parameters when creating a new VM. **Or we can get vm1 to boot by running the first playbook one more time**, since this time the task will be interpreted as attempting to change the state of an existing VM to running, which is what we want." 

I interpreted this to mean that the playbook could remain unchanged and it would still start the VM. Am I misreading it? Thanks for your help, @mmazur

Reply:
If you take a look at the [examples for that blog post](https://github.com/kubevirt/ansible-kubevirt-modules/tree/master/examples/blog/part1):

1. the first one basically just does `state: running`,
2. while the second one actually defines the vm.

And when I wrote that _we can get vm1 to boot by running the **first** playbook one more time_, that means the one with `state: running`. Re–running the **second** example (the one with the vm spec) will do nothing, unless you've added `state: running` to it first.

So with your text describing only a single playbook, it needs to be made explicitly clear, that to make the vm actually boot, the playbook being run needs to contain `state: running`. As in, something like this:

_6. If you did not include `state: running` in your playbook file, you can boot the virtual machine by adding that line to the playbook file and running the playbook again:_

Reply:
Ah, thanks so much for clarifying! That makes much more sense. I will make that change shortly.